src/itehandler/IteToSwitch.h

```diff
@@ -0,0 +1,177 @@
+//
+// Created by prova on 04.08.20.
+//
+
+#ifndef OPENSMT_ITETOSWITCH_H
+#define OPENSMT_ITETOSWITCH_H
+
+#include "Logic.h"
+
+
+namespace ite {
+    class timer {
+        time_t start;
+        const std::string name;
+    public:
+        explicit timer(std::string&& name) : start(time(nullptr)), name(name) {}
+        ~timer() { std::cout << name << ": " << time(nullptr) - start << std::endl; }
+    };
+
+    struct NodeRef {
+        uint32_t x;
+        inline friend bool operator== (NodeRef a1, NodeRef a2)   { return a1.x == a2.x; }
+        inline friend bool operator!= (NodeRef a1, NodeRef a2)   { return a1.x != a2.x; }
+        inline friend bool operator< (NodeRef a1, NodeRef a2)    { return a1.x < a2.x;  }
+    };
+    struct NodeRef_lt {
+        bool operator() (const NodeRef &n1, const NodeRef &n2) const { return n1 < n2; }
+    };
+    static struct NodeRef NodeRef_Undef = {INT32_MAX};
+
+    struct NodeRefHash {
+        uint32_t operator () (const PTRef& s) const {
```

Comment:
Why is `NodeRefHash` taking `PTRef` as an argument?

Reply:
Seems to be dead code anyway.